### DISCUSSION TITLE
Migrate domain

### DISCUSSION THREAD

**Discussion by neroita** [Ylianst/MeshCentral#5485] [General]
Hi all , 
I have a meshcentral 1.1.14 perfectly running on my domain that receive comunication from agents on domain1:10000 .
I would like to move the service to domain2:10000 .

I can simply publish port to the second domain but , how can I inform all agent to switch to the new domain ?
The old dns name need to be removed so only the new domain will remain.

thanks in advance,
marco.

**Comment by si458** [comment]
Sadly this isn't possible,
You can try changing the url inside meshagent.msh that sits alongside ur exe in program files
But from experience this doesn't work
So the only thing u can do is customise a new agent so it's different to the first one with ur new domain, install the new agent remotely, then uninstall the old agent

**Reply by neroita** [reply]
ok , can I remotely upgrade the agent overwriting the old one ?

**Reply by si458** [reply]
I wouldn't recommend this, as anything can happen! But you can give it a ago?

I would use a test machine u have physical access to or at least another remote support, to control it with if u lose access to it

Trial and error really, let us no how u get on tho and what steps u took if u get it working as could add it to docs!

**Reply by neroita** [reply]
ok , I have a month to do that so will try with some test machine :-D

**Reply by kameleon25** [reply]
We did similar a couple of years ago when we moved from the public meshcentral site to a self hosted one. Basically we used our PDQ Deploy software to push the files to the client then execute a batch file. This batch file removed the agent connection to the public site and installed the self hosted version. It worked a treat back then so I have no reason to believe it wouldn't work now.
